test/extended/authentication/oidc.go

```diff
@@ -572,7 +572,30 @@ func waitForRollout(ctx context.Context, client *exutil.CLI) {
 		gomega.Expect(nipCond.Status).To(o.Equal(operatorv1.ConditionTrue), "NodeInstallerProgressing condition should be True", nipCond)
 	}).WithTimeout(10*time.Minute).WithPolling(20*time.Second).Should(o.Succeed(), "should eventually begin rolling out a new revision")
 
+	// Then wait for KAS NodeInstallerProgressing condition to flip to "False".
+	// This means that the KAS-O has successfully been rolled out
+	// with our auth resource changes.
+	o.Eventually(func(gomega o.Gomega) {
+		kas, err := kasCli.Get(ctx, "cluster", metav1.GetOptions{})
+		gomega.Expect(err).NotTo(o.HaveOccurred(), "should not encounter an error fetching the KAS")
+
+		found := false
+		nipCond := operatorv1.OperatorCondition{}
+		for _, cond := range kas.Status.Conditions {
+			if cond.Type == condition.NodeInstallerProgressingConditionType {
+				found = true
+				nipCond = cond
+				break
+			}
+		}
+
+		gomega.Expect(found).To(o.BeTrue(), "should have found the NodeInstallerProgressing condition")
+		gomega.Expect(nipCond.Status).To(o.Equal(operatorv1.ConditionFalse), "NodeInstallerProgressing condition should be False", nipCond)
+	}).WithTimeout(20*time.Minute).WithPolling(10*time.Second).Should(o.Succeed(), "should eventually begin rolling out a new revision")
```

Comment:
Does 20 minutes seem reasonable here? I was debating using 30 minutes in case we get a slow rollout, but anything larger than 30 seems a bit too long

Reply:
30m would feel more appropriate IMHO, since the KAS rollout is the longest part of this stage I think. I would also consider decreasing the operator status wait to 20m.